src/server/scripts/Custom/SkipDK.cpp

```diff
@@ -158,8 +162,145 @@ class Trinitycore_skip_deathknight : public PlayerScript
+            CloseGossipMenuFor(player);
+
+            switch (gossipListId)
+            {
+            case 11:
+                AddGossipItemFor(player, GOSSIP_ICON_INTERACT_1, "Yes", GOSSIP_SENDER_MAIN, 12);
+                AddGossipItemFor(player, GOSSIP_ICON_INTERACT_1, "No", GOSSIP_SENDER_MAIN, 13);
+                SendGossipMenuFor(player, player->GetGossipTextId(me), me->GetGUID());
+                break;
+
+            case 12:
+                if (player->GetLevel() <= DKL)
+                {
+                    player->SetLevel(DKL);
+                }
+                player->LearnSpell(53428, false);//runeforging
+                player->LearnSpell(53441, false);//runeforging
+                player->LearnSpell(53344, false);//runeforging
+                player->LearnSpell(62158, false);//runeforging
+                player->LearnSpell(33391, false);//journeyman riding
+                player->LearnSpell(54586, false);//runeforging credit
+                player->LearnSpell(48778, false);//acherus deathcharger
+                player->LearnSkillRewardedSpells(776, 375);//Runeforging
+                player->LearnSkillRewardedSpells(960, 375);//Runeforging
+                player->EquipNewItem(EQUIPMENT_SLOT_HEAD, 38661, true);//Greathelm of the Scourge Champion
+                player->EquipNewItem(EQUIPMENT_SLOT_WRISTS, 38666, true);//Plated Saronite Bracers
+                player->EquipNewItem(EQUIPMENT_SLOT_WAIST, 38668, true);//The Plaguebringer's Girdle
+                player->EquipNewItem(EQUIPMENT_SLOT_HANDS, 38667, true);//Bloodbane's Gauntlets of Command
+                player->EquipNewItem(EQUIPMENT_SLOT_CHEST, 38665, true);//Saronite War Plate
+                player->EquipNewItem(EQUIPMENT_SLOT_LEGS, 38669, true);//Engraved Saronite Legplates
+                player->EquipNewItem(EQUIPMENT_SLOT_SHOULDERS, 38663, true);// Blood-Soaked Saronite Plated Spaulders
+                player->EquipNewItem(EQUIPMENT_SLOT_FEET, 38670, true);//Greaves of the Slaughter
+                player->EquipNewItem(EQUIPMENT_SLOT_TRINKET1, 38675, true);//Signet of the Dark Brotherhood
+                player->EquipNewItem(EQUIPMENT_SLOT_TRINKET2, 38674, true);//Soul Harvester's Charm
+                player->EquipNewItem(EQUIPMENT_SLOT_FINGER1, 38671, true);//Valanar's Signet Ring
+                player->EquipNewItem(EQUIPMENT_SLOT_FINGER2, 38672, true);// Keleseth's Signet Ring
+                player->AddItem(39320, true);//Sky Darkener's Shroud of Blood
+                player->AddItem(38664, true);//Sky Darkener's Shroud of the Unholy
+                player->AddItem(39322, true);//Shroud of the North Wind
+                player->AddItem(38632, true);//Greatsword of the Ebon Blade
+                player->AddItem(6948, true);//Hearthstone
+                player->AddItem(38707, true);//Runed Soulblade
+                player->AddItem(40483, true);//Insignia of the Scourge
```

Comment:
What happens if the player has no free space ? Are items lost or emailed automatically ? Is that good enough or should it be modified ? In this PR or in another PR ?

Reply:
There is plenty of space as the DK class starts off with 4 bags and the starter bag. Only class in wotlk that starts off with five bags total. and the items do not max out the inventory space of the main player bag. The items will not be lost out side of the basic armor upgrade that every DK player will have when they are done with the starter area. The option to optional skip will only be unique to the first lich king npc which will only be present for about 3 quests, after that the Option to skip the DK becomes absent from the player. This is as intended by design.